### PR TITLE
feat: save location name in a shared preference

### LINKED ISSUE #159: Save location name in Home Screen
The name of the location should be saved in a shared Preference so that the user doesn't have to enter the same location again and events can be fetched automatically

Working on it

### PR DESCRIPTION
**Pull request by nikit19**
fixes #159 
Now events can be loaded automatically when the user opens the app
ezgif com-video-to-gif 8

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt [hunk 1]
```diff
@@ -19,12 +19,15 @@ import android.widget.TextView
...
 import kotlinx.android.synthetic.main.content_no_internet.view.*
+import org.fossasia.openevent.general.data.Preference
 
 
 class EventsFragment : Fragment() {
     private val eventsRecyclerAdapter: EventsRecyclerAdapter = EventsRecyclerAdapter()
     private val eventsViewModel by viewModel<EventsViewModel>()
     private lateinit var rootView: View
+    private val preference: Preference = Preference()
+    private  val tokenKey = "LOCATION"
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: There's a reason to use MVVM. Please don't add logic in Views

File: app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt [hunk 2]
```diff
@@ -19,13 +19,15 @@ import android.widget.TextView
...
 import kotlinx.android.synthetic.main.content_no_internet.view.*
+import org.fossasia.openevent.general.data.Preference
 import org.fossasia.openevent.general.utils.Utils
 
 
 class EventsFragment : Fragment() {
     private val eventsRecyclerAdapter: EventsRecyclerAdapter = EventsRecyclerAdapter()
     private val eventsViewModel by viewModel<EventsViewModel>()
     private lateinit var rootView: View
+    private val preference: Preference = Preference()
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: No logic in fragment

Comment (nikit19) [reply app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: @iamareebjamal I moved the logic viewmodel in this commit https://github.com/fossasia/open-event-android/pull/172/commits/0636e281fd003e7976388320f3091bc858a825a8
but the tests were failing so I reverted.
Was it the correct way?

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: Yes

**Comment by nikit19** [comment]
@iamareebjamal I tried to fix the failing dependency test but I wasn't able
`org.koin.error.BeanInstanceCreationException: Can't create bean Factory[class=org.fossasia.openevent.general.event.EventsViewModel, binds~(android.arch.lifecycle.ViewModel)] due to error : java.lang.NullPointerException`

Could you please help

File: app/src/main/java/org/fossasia/openevent/general/event/EventsViewModel.kt [hunk 3]
```diff
@@ -17,9 +17,10 @@ class EventsViewModel(private val eventService: EventService) : ViewModel() {
     val events = MutableLiveData<List<Event>>()
     val error = MutableLiveData<String>()
 
-    var locationName: String? = null
+    var locationName: String? = preference.getString(tokenKey)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/EventsViewModel.kt]: Make this lazy, it'll work

File: app/src/main/java/org/fossasia/openevent/general/event/EventsViewModel.kt [hunk 4]
```diff
@@ -17,9 +17,10 @@ class EventsViewModel(private val eventService: EventService) : ViewModel() {
...
 
     fun loadLocationEvents() {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/EventsViewModel.kt]: Why not take the location name as parameter

**Comment by nikit19** [comment]
@iamareebjamal Did all the requested changes please review

File: app/src/main/java/org/fossasia/openevent/general/event/EventsViewModel.kt [hunk 5]
```diff
@@ -5,19 +5,23 @@ import android.arch.lifecycle.ViewModel
...
 import io.reactivex.schedulers.Schedulers
+import org.fossasia.openevent.general.data.Preference
+import org.fossasia.openevent.general.utils.nullToEmpty
 import timber.log.Timber
 
-class EventsViewModel(private val eventService: EventService) : ViewModel() {
+class EventsViewModel(private val eventService: EventService, private val preference: Preference) : ViewModel() {
 
     private val compositeDisposable = CompositeDisposable()
+     val tokenKey = "LOCATION"
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/EventsViewModel.kt]: private val

**Comment by nikit19** [comment]
PR updated, please review

**Pull request merged**